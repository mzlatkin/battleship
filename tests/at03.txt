-- student test 3
debug_test(easy)
bomb([A,1],[A,2])
bomb([C,1],[C,2])
undo
undo
redo
