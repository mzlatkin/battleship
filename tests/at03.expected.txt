  state 0 OK -> Start a new game
->debug_test(easy)
  state 1 OK -> Fire Away!
     1  2  3  4
  A  _  _  _  _
  B  _  _  _  v
  C  _  _  h  v
  D  _  _  _  _
  Current Game (debug): 1
  Shots: 0/8
  Bombs: 0/2
  Score: 0/3 (Total: 0/3)
  Ships: 0/2
    2x1: [B, 4]->v;[C, 4]->v
    1x1: [C, 3]->h
->bomb([A, 1],[A, 2])
  state 2 OK -> Miss! Keep Firing!
     1  2  3  4
  A  O  O  _  _
  B  _  _  _  v
  C  _  _  h  v
  D  _  _  _  _
  Current Game (debug): 1
  Shots: 0/8
  Bombs: 1/2
  Score: 0/3 (Total: 0/3)
  Ships: 0/2
    2x1: [B, 4]->v;[C, 4]->v
    1x1: [C, 3]->h
->bomb([C, 1],[C, 2])
  state 3 OK -> Miss! Keep Firing!
     1  2  3  4
  A  O  O  _  _
  B  _  _  _  v
  C  O  O  h  v
  D  _  _  _  _
  Current Game (debug): 1
  Shots: 0/8
  Bombs: 2/2
  Score: 0/3 (Total: 0/3)
  Ships: 0/2
    2x1: [B, 4]->v;[C, 4]->v
    1x1: [C, 3]->h
->undo
  state 4 (= state 2) OK -> Miss! Keep Firing!
     1  2  3  4
  A  O  O  _  _
  B  _  _  _  v
  C  _  _  h  v
  D  _  _  _  _
  Current Game (debug): 1
  Shots: 0/8
  Bombs: 1/2
  Score: 0/3 (Total: 0/3)
  Ships: 0/2
    2x1: [B, 4]->v;[C, 4]->v
    1x1: [C, 3]->h
->undo
  state 5 (= state 1) OK -> Fire Away!
     1  2  3  4
  A  _  _  _  _
  B  _  _  _  v
  C  _  _  h  v
  D  _  _  _  _
  Current Game (debug): 1
  Shots: 0/8
  Bombs: 0/2
  Score: 0/3 (Total: 0/3)
  Ships: 0/2
    2x1: [B, 4]->v;[C, 4]->v
    1x1: [C, 3]->h
->redo
  state 6 (= state 2) OK -> Miss! Keep Firing!
     1  2  3  4
  A  O  O  _  _
  B  _  _  _  v
  C  _  _  h  v
  D  _  _  _  _
  Current Game (debug): 1
  Shots: 0/8
  Bombs: 1/2
  Score: 0/3 (Total: 0/3)
  Ships: 0/2
    2x1: [B, 4]->v;[C, 4]->v
    1x1: [C, 3]->h
